### PR TITLE
Implement the service account "flow"

### PR DESCRIPTION
**Pull request by dermesser**
Sorry for the delay, I finally got around to doing it :)

tl;dr: The service account flow doesn't really fit the Authenticator model (because it's non-interactive), so I made it a distinct implementation of `GetToken`.

I also initially planned on using the `jwt` crate. However, that crate sends a JWT header of `{"type": "RSS256", "typ": "JWT", "kid": null}`, and the last `kid` property causes the OAuth frontend to return a `500`. I fell back to implementing the functionality myself, i.e. signing the message using OpenSSL (as it turns out, I had to use OpenSSL anyways in order to parse the key for `jwt` to sign the message...).

There's also the `frank_jwt` crate, but looking through it I wasn't sure whether it'd work smoothly with the picky Google frontend.

Let me know what you think!

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing. Unfortunately I have to tell you that there were the following issues with your Pull Request

 * Commit: 36d38157477a1a40f474cfe673345da1fdcc8d99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: cd5e31e38c4e923c39f2763bb1a1289f39ab8f50
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

 * Commit: 8522a959d82b732cbda86c2a96632663cca7c03e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by dermesser** [comment]
also, consider re-generating the documentation after merge :)

**Comment by GitCop** [comment]
Thanks for contributing. Unfortunately I have to tell you that there were the following issues with your Pull Request

 * Commit: 5f86047672212e926745e2b380cb9be768b5b593
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by Byron** [comment]
I can't find the words to express my gratitude ! You are by now single-handedly driving this project forward, and in a way most of what you are submitting these days is beyond me ! It would need time for a proper review, and I assume the tests are all that's needed to at least verify the 'happy-path'.

Taken that into consideration, I'd love you to become a collaborator, granting you full write access to this repository. Given that most of the usable flows are made by your hand, that's the least I can do.

**Comment by Byron** [comment]
@dermesser Regarding the docs, those are rebuilt automatically from the latest master. I could just verify that it does indeed (still) work.

**Comment by dermesser** [comment]
Thanks, I gladly accepted the invitation! :)
